src/wallet/wallet.cpp

```diff
@@ -2042,6 +2054,9 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, int nConfMine, int
         if (output.nDepth < (pcoin->IsFromMe(ISMINE_ALL) ? nConfMine : nConfTheirs))
             continue;
 
+        if (pcoin->InMempool() && std::max(pcoin->GetMempoolAncestorCount(), pcoin->GetMempoolDescendantCount()) >= nMaxAncestors)
```

Comment:
Why would we compare the descendant count to the max *ancestors*? Shouldn't it be compared to max descendants?